### DISCUSSION TITLE
How to write tests for a project like this ?

### DISCUSSION THREAD

**Discussion by aahnik** [ramnes/notion-sdk-py#52] [Q&A]
Tests will involve hitting the Notion API. An API Key is also required. How the tests can be standardized across all the people?

API key could be kept in github secrets. but other collaborators will not get it, so they cant run those same tests locally.

How to make sure that everyone runs same tests with the same data ?

One thing is to save the data returned by the API in a json file, and commit that to the source code. I hope something like that is done by vcrpy ? Should the cassete be committed to source control ?

**Comment by guedes** [comment]
> Should the cassete be committed to source control ?

IMHO, all test's mocked data should be source-controlled.
